src/NServiceBus.Hosting.Windows/StartableAndStoppable/StartableAndStoppableRunner.cs

```diff
@@ -21,9 +22,13 @@ public Task Start(IMessageSession session)
             var startableTasks = new List<Task>();
             foreach (var startable in wantToRunWhenBusStartsAndStops)
             {
-                var task = startable.Start(session).ThrowIfNull();
-
                 var startable1 = startable;
+                var startableName = startable1.GetType().AssemblyQualifiedName;
+
+                var longRunningTokenSource =
+                    LongRunningWarning($"The start method for {startableName} is taking a long time to complete. The endpoint will not start until this operation has completed.");
```

Comment:
I wonder whether it would be cleaner to just start a new task after `startable1.Start()` checking the status of `task` or using Task.WhenAny instead of a CTS which needs to be handled in both continuations.

e.g. something like

```
var task = startable1.Start(session).ThrowIfNull();
Task.Delay(TimeSpan...).ContinueWith(t => 
{
    if(!(task.Status.IsCompleted || task.IsFaulted || task.IsCanceled))
    {
     Log.Warn(...)
    }
}
```
var task = startable1.Start(session).ThrowIfNull();
or 

```
var task = startable1.Start(session).ThrowIfNull();
var warningWaitTimeTask = Task.Delay(TimeSpan...);
Task.WhenAny(task, warningWaitTimeTask).ContinueWith(t => 
{   
 if(t == warningWaitTimeTask)
    {
          Log.Warn(....)
     }
});
```

/cc @danielmarbach